### PR TITLE
samples: matter: make device types conform with the spec

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Update ZAP configuration of Matter samples so that it includes all required cluster and cluster elements for given device types. Also, align the root endpoint configuration with Matter Weather Station which passed Matter 1.0 SVE.

The change depends on and includes https://github.com/nrfconnect/sdk-nrf/pull/9002. Turn off displaying the first commit when reviewing.

### CODE REVIEW

File: applications/matter_weather_station/README.rst [hunk 1]
```diff
@@ -13,7 +13,8 @@ The device works as a Matter accessory device, meaning it can be paired and cont
 You can use this application as a reference for creating your own application.
 
 .. note::
-    The Matter protocol is in an early development stage and must be treated as an :ref:`experimental <software_maturity>` feature.
+   The Thingy:53 Matter weather station application featured in the |NCS| v2.1.1 release participated in Matter Specification Validation Event (SVE) and successfully passed all required test cases to be considered as a device compliant with Matter 1.0.
+   You can use |NCS| v2.1.1 release to see the configuration and files that were originally used in Matter 1.0 certification.
```
Comment (greg-fer) [inline applications/matter_weather_station/README.rst:17]: Apply edits to this file as discussed in https://github.com/nrfconnect/sdk-nrf/pull/9002 from @kkasperczyk-no

**Review by greg-fer — DISMISSED** [review]

File: applications/matter_weather_station/README.rst (hunk 1 above)
Comment (kkasperczyk-no) [reply applications/matter_weather_station/README.rst:17]: @greg-fer I believe this PR will be rebased after merging #9002 and my commit was pulled only to provide some base for testing Damian changes, so maybe modifying it is not needed.

Comment (greg-fer) [reply applications/matter_weather_station/README.rst:17]: Ah ok, mysteries of the GitGods!

**Review by greg-fer — COMMENTED** [review]
Just make sure to rebase to include changes from @kkasperczyk-no 's PR. I don't think my approval is needed, as these doc changes were approved elsewhere.

File: applications/matter_weather_station/README.rst (hunk 1 above)
Comment (Damian-Nordic) [reply applications/matter_weather_station/README.rst]: Rebased. Please re-review. @greg-fer - yes, I guess I still need your approval although you already reviewed it on the other branch.

**Review by kkasperczyk-no — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**